File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.cpp
class Solution {
public:
    long long maximumHappinessSum(vector<int>& happiness, int k) {
        long long sum = 0;

        sort(happiness.begin(), happiness.end(), greater<int>());

        for(int i=0; i<k; i++){
            happiness[i] -= i;
            if(happiness[i] > 0) sum += happiness[i];
        }

        return sum;
    }
};

File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        sum = 0
        happiness.sort(reverse = True)

        for i in range(k):
            happiness[i] -= i
            if(happiness[i] > 0):
                sum += happiness[i]
        
        return sum
